File: algorithms/palindrome/palindrome.js
function iterativePalindrome(str) {
  for (let i = 0; i < str.length; i++) {
    if (str[i] !== str[str.length - 1 - i]) {
      return false;
    } else return true;
  }
}

function recursivePalindrome(str) {
  if (str.length <= 1) {
    return true;
  } else {
    if (str[0] === str[str.length - 1]) {
      str = str.substr(1, str.length - 2);
      return recursivePalindrome(str);
    }
  }
  return false;
}

const result1 = iterativePalindrome('racecar');
const result2 = iterativePalindrome('notapalindrome');
console.log(result1, result2);

const result3 = recursivePalindrome('racecar');
const result4 = recursivePalindrome('notapalindrome');
console.log(result3, result4);

File: algorithms/palindrome/palindrome.py
def iterative_palindrome(str):
    for i in range(len(str)):
        if str[i] != str[len(str) - 1 - i]:
            return False
        else: return True

def recursive_palindrome(str):
    if len(str) <= 1:
        return True
    else:
        if str[0] == str[len(str)-1]:
            str = str[1:-1]
            return recursive_palindrome(str)
    return False


result1 = iterative_palindrome("racecar")
result2 = iterative_palindrome("notapalindrome")
print(result1, result2)

result3 = recursive_palindrome("racecar")
result4 = recursive_palindrome("notapalindrome")
print(result3, result4)